files/default/neutron-ha-tool.py

```diff
@@ -654,10 +654,14 @@ def migrate_router(qclient, router_id, agent, target,
     qclient.remove_router_from_l3_agent(agent['id'], router_id)
     LOG.debug("Removed router from agent=%s" % agent['id'])
 
-    # ensure it is removed or log an error
+    # ensure it is removed
     if router_id in list_routers_on_l3_agent(qclient, agent['id']):
-        raise RuntimeError("Failed to remove router_id=%s from agent_id=%s" %
-                           (router_id, agent['id']))
+        qclient.remove_router_from_l3_agent(agent['id'], router_id)
+        LOG.debug("Removed (again) router from agent=%s" % agent['id'])
```

Comment:
@aplanas I don't like the workaround neither but the fix seems far down the road, because heavy backport is needed.

The retry mechanism you mention does not solve the problem here. The first (and incomplete) delete does not produce an error thus the retry mechanism is not activated.

About your two comments:
- I'll add a reference to the bsc#, good point.
- I can change the output on the retry. I like the new text you proposed. I don't think though it's a warning since it's expected behavior for cloud6.